openlibrary/plugins/upstream/account.py

```diff
@@ -431,6 +437,51 @@ def POST_resend_verification_email(self, i):
         )
         return render.message(title, message)
 
+    def set_screener_cookie(self, account: OpenLibraryAccount):
+        if self.is_eligible_for_screener(account):
+            # `se` is "Survey eligible"
+            web.setcookie('se', '1', expires=(3600 * 24 * 30))  # Expires in 30 days
+
+    def is_eligible_for_screener(self, account: OpenLibraryAccount) -> bool:
+        # Current month must be August:
+        if (now := datetime.now()) and now.month != 8:
+            return False
+
+        # Account must be at least 90 days old:
+        if days_since(account.creation_time()) < 90:
+            return False
+
+        # Account was created using a university's domain:
+        email = account.email
+        if not self.is_edu_domain(email):
+            return False
+
+        # Has borrowed at least three books:
+        if not self.has_borrowed_at_least(3, account.s3_keys):
+            return False
+
+        return True
+
+    def is_edu_domain(self, email: str) -> bool:
+        if not email or '@' not in email:
+            return False
+
+        domain = email.split('@')[-1]
+
+        sorted_edu_domains = utils.get_edu_domains()
+
+        i = bisect_left(sorted_edu_domains, domain)
+        if i != len(sorted_edu_domains) and sorted_edu_domains[i] == domain:
+            return True
```

Comment:
We may be able to use a set instead of a list and get constant time lookup

Reply:
Not able to cache sets currently, so have used a `dict` instead of a `set`.